contracts/veLON.sol

```diff
@@ -0,0 +1,347 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.7.6;
+
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721Metadata.sol";
+
+import "./interfaces/IveLON.sol";
+import "./interfaces/ILon.sol";
+import "./Ownable.sol";
+
+contract veLON is IveLON, ERC721, Ownable, ReentrancyGuard {
+    using SafeMath for uint256;
+
+    uint256 private constant WEEK = 1 weeks;
+    uint256 public constant PENALTY_RATE_PRECISION = 10000;
+    uint256 internal constant MULTIPLIER = 1 ether;
+    address public immutable token;
+
+    uint256 public tokenSupply;
+    uint256 public epoch;
+    uint256 public maxLockDuration = 365 days;
+    uint256 public earlyWithdrawPenaltyRate = 3000;
+
+    mapping(uint256 => Point) public poolPointHistory; // epoch -> unsignd point
+    mapping(uint256 => Point[1000000000]) public userPointHistory; // user -> Point[user_epoch]
+    mapping(uint256 => LockedBalance) public locked; // tokenId -> locked balance
+    mapping(uint256 => uint256) public userPointEpoch; // tokenId -> epoch
+    mapping(uint256 => int256) public slope_changes; // time -> signed slope change
```

Comment:
We might change this to `slopeChange`

Reply:
Actually I left it with underscore because I was thinking of updating couple namings in this contract. But let's fix the case first. Fixed in 9a24720.